partiql-spi/src/main/kotlin/org/partiql/spi/fn/builtins/AggAvg.kt

```diff
@@ -7,158 +7,139 @@ import org.partiql.spi.fn.Agg
 import org.partiql.spi.fn.AggSignature
 import org.partiql.spi.fn.FnParameter
 import org.partiql.spi.fn.builtins.internal.AccumulatorAvg
-import org.partiql.value.PartiQLValueExperimental
-import org.partiql.value.PartiQLValueType.ANY
-import org.partiql.value.PartiQLValueType.DECIMAL
-import org.partiql.value.PartiQLValueType.DECIMAL_ARBITRARY
-import org.partiql.value.PartiQLValueType.FLOAT32
-import org.partiql.value.PartiQLValueType.FLOAT64
-import org.partiql.value.PartiQLValueType.INT
-import org.partiql.value.PartiQLValueType.INT16
-import org.partiql.value.PartiQLValueType.INT32
-import org.partiql.value.PartiQLValueType.INT64
-import org.partiql.value.PartiQLValueType.INT8
-
-@OptIn(PartiQLValueExperimental::class)
+import org.partiql.types.PType
+
 internal object Agg_AVG__INT8__INT8 : Agg {
 
     override val signature: AggSignature = AggSignature(
         name = "avg",
-        returns = DECIMAL,
+        returns = PType.typeDecimalArbitrary(),
```

Comment:
It may be worth adding a comment

> If AVG is specified and DT is exact numeric, then the declared type of the result is exact numeric with implementation-defined precision not less than the precision of DT and implementation-defined scale not less than the scale of DT. SQL-99 p.156